### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by habasm**
### Objectives
- Understand the concept of state management.
- Use store, actions, and reducers in React.

### Actions
- Setup a Redux store
- Create a slice to save your state and reducers

### CODE REVIEW

**Comment by TedLivist** [comment]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @habasm,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### - Your README contains the Microverse logo. Kindly remove it as part of the customization requirement.
### - You should have a Future Features section listing one or more features you intend to implement in the future. Ensure there is a menu link for it as well.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by habasm** [comment]
### @TedLivist thanks, now it's fixed.

**Comment by TedLivist** [comment]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @habasm,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You still do not have the "Future Features" section as pointed out. Kindly refer to [the template](https://github.com/microverseinc/readme-template) for guidance.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by habasm** [comment]
@TedLivist Thanks again, I tried to fixed it also.

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    case REMOVE_BOOKS:
+      return {
+        books: [...state.books.filter((book) => book.Id !== action.id)],
+      };
+    default:
+      return state;
+  }
+};
+
+export default bookReducer;
```
Comment (FrederickMih) [inline src/redux/books/books.js]: - [ ] Hi @habasm, once more, let's kindly adhere to the requirements of this project. Let's kindly import the necessary methods from Redux Toolkit as requested and define slice of state for books and categories. **Hints** `import { createSlice } from "@reduxjs/toolkit"` Next, you'll kindly Create an initialState with a state value books that contains an empty array. **Hints**
```
initialState: {
    books: [],
  }
```

- After creating two reducers addBook and removeBook export by default bookReducer.reducer. Reducers should be created inside:

```
export const bookReducer = createSlice({

})
```

- Kindly check the requirements once more below on the screenshot:

image

File: src/components/Categories/Categories.js [hunk 2]
```diff
@@ -1,10 +1,21 @@
...
+
+  return (
+    <div>
+      <h2>{categories.categories[0]}</h2>
+      <button type="button" onClick={onClickHandler}>Check Status</button>
+    </div>
+  );
+};
 
 export default Categories;
```
Comment (FrederickMih) [inline src/components/Categories/Categories.js]: - [ ] Kindly check the comment above :arrow_up:

File: README.md [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 
   <img src="/murple_logo.png" alt="logo" width="140"  height="auto" />
   <br/>
 
   <h3><b>Microverse</b></h3>
 
-</div>
+</div> -->
 <div align="left">
   <h1>The Bookstore CMS</h1>
```
Comment (FrederickMih) [inline README.md]: - [**Optional**] @habasm your README.md file contains two project titles. Below and above table of content.
Screenshot from 2023-02-28 09-25-50

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand:  @habasm , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- PR is professional :+1: 
- Linter checks passed :heavy_check_mark: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by christianbiring1 — APPROVED** [review]
Hi @habasm , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
# STATUS: APPROVED :tada: 
nicee

## Highlights
- [ ] All linters checks are green :heavy_check_mark: 
- [ ] The redux is set properly to manage the app state :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**